### PR TITLE
[DOCS-6021] Update Onboarding Experience w/ "Sending Traces to Datadog"

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do?

- ~Move language-specific pages into tabs on Sending Traces to Datadog.~
- Add language-agnostic Agent installation steps to main page.
- Move `Inject the library locally without modifying your application code` to Option 1 in Step 2.
- Remove mention of "remote" injection. Replace with "in the UI".

### Motivation
@vidurkhannadatadog and I are working on improving the onboarding documentation experience.

[DOCS-6021](https://datadoghq.atlassian.net/browse/DOCS-6021)

### Preview
https://docs-staging.datadoghq.com/brett0000FF/DOCS-6021/tracing/trace_collection/

### Additional Notes

There are a lot of commits in this PR because we initially experimented with some other approaches.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-6021]: https://datadoghq.atlassian.net/browse/DOCS-6021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/tracing/trace_collection/_index.md [hunk 1]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
 
-For setup instructions, select your language:
+### Option 1 - Inject the library locally without modifying your application code
 
-{{< partial name="apm/apm-languages.html" >}}
+For Kubernetes, hosts, and containers, you can inject the tracing library locally without modifying your application code. For more information and instructions, read [Injecting Libraries Locally][2].
 
-<br>
+- For Kubernetes, you can inject the library into applications written in Java, Python, Node.js, .NET, and Ruby.
+- **Beta**: For Linux hosts and containers, you can inject the library into applications written in Java, Python, Node.js, and .NET.
```
Comment (wantsui) [inline content/en/tracing/trace_collection/_index.md:103]: The Library Injection docs for Ruby say that it's in Beta: https://docs.datadoghq.com/tracing/trace_collection/library_injection_local/?tab=kubernetes#requirements 

> For Ruby, library injection support is in Beta. Instrumentation is only supported for Ruby on Rails or Hanami applications at this time.

**Review by wantsui — COMMENTED** [review]
The restructure and additions looks good. I did leave a question about Library Injection support since there was a difference comparing against this doc: https://docs.datadoghq.com/tracing/trace_collection/library_injection_local/?tab=kubernetes#requirements .

File: content/en/tracing/trace_collection/_index.md (hunk 1 above)
Comment (brett0000FF) [reply content/en/tracing/trace_collection/_index.md:103]: Good catch! @vidurkhannadatadog -- Should we indicate that library injection for Ruby is in beta?

File: content/en/tracing/trace_collection/_index.md [hunk 2]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
-Depending on the programming language and infrastructure you use, you have the following options to instrument your application.
+Depending on the programming language and infrastructure you use, you have the following options to instrument your application:
 
-### Option 1 - Add the library directly to your application code
+- [Inject the library locally without modifying your application code](#option-2---inject-the-library-locally-without-modifying-code)
+- [Add the library directly to your application code](#option-1---add-the-library-directly-to-your-application-code)
+- [Inject the library from the Datadog UI](#option-3---inject-the-library-from-the-datadog-ui)
 
-For setup instructions, select your language:
+### Option 1 - Inject the library locally without modifying your application code
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:98]: This is very wordy, can we just say "Auto-instrument your application"?

File: content/en/tracing/trace_collection/_index.md [hunk 3]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
 
-{{< partial name="apm/apm-languages.html" >}}
+For Kubernetes, hosts, and containers, you can inject the tracing library locally without modifying your application code. For more information and instructions, read [Injecting Libraries Locally][2].
 
-<br>
+- For Kubernetes, you can inject the library into applications written in Java, Python, Ruby (Beta), Node.js, and .NET.
+- **Beta**: For Linux hosts and containers, you can inject the library into applications written in Java, Python, Node.js, and .NET.
 
-To instrument an application written in a language that does not have official library support, see the list of [community tracing libraries][2].
+### Option 2 - Add the library directly to your application code
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:105]: The wording is confusing, can we say "Manually instrument your application"?

File: content/en/tracing/trace_collection/_index.md [hunk 4]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
-For Kubernetes, hosts, and containers, you can inject the tracing library locally at the Agent without modifying your application code. For more information and instructions, read [Injecting Libraries Locally][4].
+{{< partial name="apm/apm-languages.html" >}}
 
-- For Kubernetes, you can inject the library into applications written in Java, Python, Node.js, .NET, and Ruby.
-- **Beta**: For Linux hosts and containers, you can inject the library into applications written in Java, Python, Node.js, and .NET.
+To instrument an application written in a language that does not have official library support, see the list of [community tracing libraries][1].
 
-### Option 3 - Inject the library remotely from the Datadog UI
 
+### Option 3 - Inject the library from the Datadog UI
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:114]: Again - can this say "Auto-instrument your application from Datadog UI"?

File: content/en/tracing/trace_collection/_index.md [hunk 5]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
-### Option 1 - Add the library directly to your application code
+- [Inject the library locally without modifying your application code](#option-2---inject-the-library-locally-without-modifying-code)
+- [Add the library directly to your application code](#option-1---add-the-library-directly-to-your-application-code)
+- [Inject the library from the Datadog UI](#option-3---inject-the-library-from-the-datadog-ui)
 
-For setup instructions, select your language:
+### Option 1 - Inject the library locally without modifying your application code
 
-{{< partial name="apm/apm-languages.html" >}}
+For Kubernetes, hosts, and containers, you can inject the tracing library locally without modifying your application code. For more information and instructions, read [Injecting Libraries Locally][2].
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:100]: I think we need to better explain what "injecting the tracing library means", something like "For Kubernetes, hosts, and containers you can auto-instrument your application by injecting the tracing library into your process."

Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:100]: I think we need to better explain what "injecting the tracing library means", something like "For Kubernetes, hosts, and containers you can auto-instrument your application by injecting the tracing library into your process."

File: content/en/tracing/trace_collection/_index.md [hunk 6]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
...
+[1]: /agent/basic_agent_usage/heroku/#installation
+[2]: /integrations/cloud_foundry/#trace-collection
+[3]: /integrations/amazon_elasticbeanstalk/
+[4]: /infrastructure/serverless/azure_app_services/#overview
+[5]: /integrations/
+[6]: /help/
+{{% /tab %}}
+{{< /tabs >}}
+
+For more language-specific details, see the [Tracing Library][4] pages.
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:84]: I think this line belongs in "Step #2"

File: content/en/tracing/trace_collection/_index.md [hunk 7]
```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
 
 If you don't use the Single Step APM Instrumentation option, APM-specific configurations are required on both the Tracer and Agent to ensure that traces can be properly received from containerized, serverless, or certain other environments. Ensure you have followed instructions for both components.
```
Comment (ik130) [inline content/en/tracing/trace_collection/_index.md:27]: We shouldn't  assume that the reader has context about Single Step APM Instrumentation.

**Review by cswatt — APPROVED** [review]

**Pull request merged**